### PR TITLE
Add Documentation for Performance yml template and scripts

### PR DESCRIPTION
**Pull request by adiaaida**

### CODE REVIEW

File: Documentation/HowToAddPerfTestingToPipeline.md [hunk 1]
```diff
@@ -0,0 +1,134 @@
+# Automate Performance testing run in AzDO Pipelines
+
+## Performance scripts
+
+The performance scripts are a powershell and shell script that set up the environment variables necessary for running performance testing. Scripts can be found [here](../eng/common/performnace).
```
Comment (chcosta) [inline Documentation/HowToAddPerfTestingToPipeline.md]: typo "performnace"

File: Documentation/HowToAddPerfTestingToPipeline.md [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+
+## How to add performance testing to a pipeline
+
+The pre-reqs for running performance testing are defined in the template -- [performance.yml](../eng/common/templates/job/performance.yml). The template will need to be added as a job to any repository that wishes to run performance testing. The only required parameter are jobName and pool. If no other parameters are supplied, the testing will pull down the performance repository and run the performance testing against the latest dotnet sdk.
```
Comment (chcosta) [inline Documentation/HowToAddPerfTestingToPipeline.md]: These links don't work (in the context of your PR), you might just use an explicit link (ie https://github.com/dotnet/arcade/blob/master/eng/common/templates/job/performance.yml)

File: Documentation/HowToAddPerfTestingToPipeline.md [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+Additional template parameters are:
+
+| Name                 | Type   | Description                               |
+| -------------------- | ------ | ----------------------------------------- |
+| displayName          | string | The name to display in the Azure Pipeline |
+| steps                | array  | Any additional steps that need to be run before running the performance testing (i.e. pulling down the build artifacts, building, etc.) |
+| container            | object | The container to run in (if the build machine is linux) |
+| extraSetupParameters | string | Extra parameters to send to the performance scripts |
+| frameworks           | array  | Dotnet frameworks to run against (i.e. netcoreapp3.0, netcoreapp2.2, etc) |
+| continueOnError      | string | Determins whether to continue the build if the step errors |
```
Comment (chcosta) [inline Documentation/HowToAddPerfTestingToPipeline.md]: typo "determins"

File: Documentation/HowToAddPerfTestingToPipeline.md [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+| dependsOn            | object | Jobs that the performance testing depends on (i.e. Previous build jobs whose aritfacts the performance testing job will use) |
```
Comment (chcosta) [inline Documentation/HowToAddPerfTestingToPipeline.md]: typo "aritfacts"

**Comment by adiaaida** [comment]
Good point.

@billwert @DrewScoggins @RussKeldorph @danmosemsft: here is the PR for the documentation for the performance testing template. Please take a look and tag anyone else who you think should take a look.

File: Documentation/HowToAddPerfTestingToPipeline.md [hunk 5]
```diff
@@ -0,0 +1,134 @@
+# Automate Performance testing run in AzDO Pipelines
+
```
Comment (danmosemsft) [inline Documentation/HowToAddPerfTestingToPipeline.md]: Perhaps you can add a bit more intro. Who is the audience here - repo owners, contributors? What does this give me - rolling perf runs? per-PR perf runs? And what artefacts do they produce?

Comment (adiaaida) [reply Documentation/HowToAddPerfTestingToPipeline.md]: Sure thing. I'll do some updating.

File: Documentation/HowToAddPerfTestingToPipeline.md [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+## Performance scripts
+
+The performance scripts are a powershell and shell script that set up the environment variables necessary for running performance testing. Scripts can be found [here](../eng/common/performance).
+
+## How to add performance testing to a pipeline
+
+The pre-reqs for running performance testing are defined in the template -- [performance.yml](../eng/common/templates/job/performance.yml). The template will need to be added as a job to any repository that wishes to run performance testing. The only required parameter are jobName and pool. If no other parameters are supplied, the testing will pull down the performance repository and run the performance testing against the latest dotnet sdk.
```
Comment (RussKeldorph) [inline Documentation/HowToAddPerfTestingToPipeline.md]: The only required parameter -> The only required parameter**s**

**Comment by adiaaida** [comment]
@chcosta I'd love to check this in, but I am seeing that all of the required statuses must pass before I can, but AzDO isn't running the checks because this path is excluded. How can I merge this?

**Pull request merged**